aci/file.go

```diff
@@ -100,3 +107,87 @@ func XzReader(r io.Reader) io.ReadCloser {
 
 	return rpipe
 }
+
+// ManifestFromImage creates a new schema.ImageManifest from the given ACI image.
+func ManifestFromImage(rs io.ReadSeeker) (*schema.ImageManifest, error) {
+	var im schema.ImageManifest
+
+	tr, err := NewCompressedTarReader(rs)
+	if err != nil {
+		return nil, err
+	}
+
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
```

Comment:
Actually this is buggy because if the reader does not have a manifest at all, then the function will return a zero-valued Imagemanifest, which is not what you want (nil)